sdk/src/main/java/me/digi/sdk/api/DMEAPIClient.kt

```diff
@@ -38,6 +38,30 @@ class DMEAPIClient(private val context: Context, private val clientConfig: DMECl
     private val httpClient: Retrofit
     internal val argonService: DMEArgonService
 
+    companion object {
+
+        fun parsedDMEError(
+            argonErrorCode: String?,
+            argonErrorMessage: String?,
+            argonErrorReference: String?
+        ) =
+            DMEAPIError::class.sealedSubclasses.fold<KClass<out DMEAPIError>, DMEAPIError?>(null) { _, err ->
+                val argonCode =
+                    (err.annotations.firstOrNull { it is ArgonCode } as? ArgonCode)?.value
+                if (argonCode == argonErrorCode) run {
+                    val instance = err.createInstance()
+                    instance.apply {
+                        code = argonErrorCode
+                        if (argonErrorMessage != null) message = argonErrorMessage
+                        reference = argonErrorReference
+                    }
+
+                    return@fold instance
+
+                } else null
+            } ?: DMEAPIError.UNMAPPED(argonErrorCode, argonErrorMessage!!, argonErrorReference)
```

Comment:
This forced unwrap makes me a little nervous, could be worth refactoring this line to `?: argonErrorMessage?.let { DMEAPIError.UNMAPPED(argonErrorCode, it, argonErrorReference) } ?: DMEAPIError.GENERIC`

Reply:
Assuming that if the value is null, we can't do any better than a generic error anyway.

Reply:
Fixed as well, all apart from empty constructors, so once I get snippet I'll push these changes and wrap this up. Or just keep empty constructors for the moment, which so far doesn't break the app which is good.